include/mgvalue.hpp

```diff
@@ -0,0 +1,1604 @@
+#pragma once
+
+#include <cstring>
+#include <initializer_list>
+#include <set>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+#include <glog/logging.h>
```

Comment:
You mustn't use `glog` in this file, this is a file that will be compiled by end users!